test/unit/remote-config/remote-config.spec.ts

```diff
@@ -543,21 +347,21 @@ describe('RemoteConfig', () => {
       const response = deepCopy(REMOTE_CONFIG_RESPONSE);
       response.conditions = Object();
       const stub = sinon
-        .stub(RemoteConfigApiClient.prototype, 'publishTemplate')
+        .stub(RemoteConfigApiClient.prototype, operationName)
         .resolves(response);
       stubs.push(stub);
-      return remoteConfig.publishTemplate(REMOTE_CONFIG_TEMPLATE)
+      return rcOperation()
         .should.eventually.be.rejected.and.have.property(
           'message', `Remote Config conditions must be an array`);
     });
 
     it('should resolve with parameters:{} when no parameters present in the response', () => {
```

Comment:
This and the subsequent `resolves` tests should not be in the `runApiResponseValidationTests` IMO. May be put them in a separate function?

Reply:
Good point. Moved to a new function `runValidResponseTests`. Renamed `runApiResponseValidationTests` to `runInvalidResponseTests`.